### ISSUE TITLE
Slow performance saving/loading images from host FS with new virtualization enabled

### ISSUE THREAD

**Issue by spmason** [docker/for-mac#6346]
- [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: 461EE3C4-9342-4D48-9828-B82E6F1256E3/20220607193519

### Expected behavior

Saving/loading images from the host FS should be comparable speeds with or without `virtualization.framework` enabled

### Actual behavior

Saving/loading images from the host FS is noticeably slower with "New virtualization framework` enabled. See reproduction for details

### Information

  - macOS Version: 12.3.1
  - Intel chip or Apple chip: Intel (MacBook Pro (16-inch, 2019))
  - Docker Desktop Version: 4.9.0 (80466)

### Output of `/Applications/Docker.app/Contents/MacOS/com.docker.diagnose check`

```
Starting diagnostics

[PASS] DD0027: is there available disk space on the host?
[PASS] DD0028: is there available VM disk space?
[PASS] DD0031: does the Docker API work?
[PASS] DD0004: is the Docker engine running?
[PASS] DD0011: are the LinuxKit services running?
[PASS] DD0016: is the LinuxKit VM running?
[PASS] DD0001: is the application running?
[PASS] DD0018: does the host support virtualization?
[PASS] DD0017: can a VM be started?
[PASS] DD0015: are the binary symlinks installed?
[PASS] DD0003: is the Docker CLI working?
[PASS] DD0013: is the $PATH ok?
[PASS] DD0007: is the backend responding?
[PASS] DD0014: are the backend processes running?
[PASS] DD0008: is the native API responding?
[PASS] DD0009: is the vpnkit API responding?
[PASS] DD0010: is the Docker API proxy responding?
[PASS] DD0012: is the VM networking working?
[PASS] DD0032: do Docker networks overlap with host IPs?
[SKIP] DD0030: is the image access management authorized?
[PASS] DD0019: is the com.docker.vmnetd process responding?
[PASS] DD0033: does the host have Internet access?
No fatal errors detected.
```

### Steps to reproduce the behavior

With virtio enabled:

```
$ time docker save debian:latest | pv > debian.tar
 123MiB 0:00:27 [4.44MiB/s]

________________________________________________________
Executed in   28.22 secs      fish           external
   usr time  446.68 millis   52.47 millis  394.20 millis
   sys time  825.00 millis   53.98 millis  771.03 millis

$ time docker load -i debian.tar
Loaded image: debian:latest

________________________________________________________
Executed in    5.58 secs      fish           external
   usr time  357.91 millis    0.14 millis  357.76 millis
   sys time  598.73 millis    1.73 millis  596.99 millis
```

With virtio disabled:

```
$ time docker save debian:latest | pv > debian.tar
 123MiB 0:00:03 [37.7MiB/s]

________________________________________________________
Executed in    3.27 secs      fish           external
   usr time  260.89 millis    0.22 millis  260.67 millis
   sys time  593.92 millis    2.17 millis  591.75 millis

$ time docker load -i debian.tar
Loaded image: debian:latest

________________________________________________________
Executed in    1.95 secs      fish           external
   usr time  236.96 millis    0.10 millis  236.86 millis
   sys time  404.98 millis    1.46 millis  403.51 millis
```

**Comment by fredericdalleau** [comment]
Thanks, we are working on the root cause, we will add this to our tests!

**Comment by djs55** [comment]
@spmason I have an early, very experimental build, which should give better performance if you'd like to try it: https://desktop-stage.docker.com/mac/main/arm64/81736/Docker.dmg

The caveats are:
1. it almost certainly requires an up-to-date macOS, it might fail to start on older versions
2. it might fail to start for other reasons, but if so you should be able to downgrade ok 
3. it's not notarized so macOS will complain and ask you to open it via Finder and confirm you're ok with this

Here's what I see on my M1:

Without virtualization.framework: (this code path is unchanged)
```
% time docker save debian:latest > debian.tar
docker save debian:latest > debian.tar  0.13s user 0.53s system 19% cpu 3.372 total
```
With the virtualization.framework: (this code path is improved)
```
% time docker save debian:latest > debian.tar     
docker save debian:latest > debian.tar  0.10s user 0.18s system 24% cpu 1.169 total
```

**Comment by spmason** [comment]
Thanks @djs55 - but I'm running on Intel not an M1. If you've got a build for that arch I can certainly give it a try

**Comment by djs55** [comment]
@spmason sorry, I should have given both links:

- https://desktop-stage.docker.com/mac/main/arm64/81736/Docker.dmg
- https://desktop-stage.docker.com/mac/main/amd64/81736/Docker.dmg

**Comment by spmason** [comment]
Hey @djs55, any update on when this fix will be released? I've checked the most recent release notes and dont see any mention of it

**Comment by djs55** [comment]
@spmason if all goes according to plan it should be released in 4.14 (the next release). So hopefully not long now!

**Comment by spmason** [comment]
Thanks @djs55 - I've installed 4.14 and re-ran the steps and the performance issues are gone now (with or without VirtIO enabled)

Thanks for all your help!
